zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py

```diff
@@ -28,3 +30,15 @@ def test_bot_help(self) -> None:
     def test_bot_no_results(self) -> None:
         with self.mock_http_conversation('test_no_result'):
             self.verify_reply('no res', 'Found no results.')
+
+    def test_attribute_error(self) -> None:
+        with self.mock_http_conversation('test_attribute_error'), \
+                patch('logging.exception'):
+            self.verify_reply('test', 'Error: Search failed. \'NoneType\' object has no attribute \'findAll\'.')
+
+    def test_bad_links(self) -> None:
```

Comment:
This test is a bit confusing to me.  If there are bad links, why does the reply say "Found Result"?

Reply:
Oh, bad links are links that should be ignored
Any cached or empty results

Reply:
It makes sure it ignores those links, then sends the zulip link which is 'under' the bad ones

Reply:
To test all the cases in:
```python
try:
            # Tries to get the href property of the URL
            link = a['href']
        except KeyError:
            continue
        # Link must start with '/url?', as these are the search result links
        if (not link.startswith('/url?')):
            continue
        # Makes sure a hidden 'cached' result isn't displayed
        if a.text.strip() == 'Cached' and 'webcache.googleusercontent.com' in a['href']:
            continue
        # a.text: The name of the page
        result = {'url': "https://www.google.com{}".format(link),
                  'name': a.text}
        results.append(result)
```

Reply:
oh, left some brackets in

Reply:
@showell Should I rename the test?

Reply:
yeah, and/or clarify in a comment what's supposed to happen